src/main/scala/rocket/ALU.scala

```diff
@@ -160,10 +162,19 @@ class ALU(implicit p: Parameters) extends AbstractALU(new ALUFN)(p) {
   val shout = Mux(io.fn === aluFn.FN_SR || io.fn === aluFn.FN_SRA, shout_r, 0.U) |
               Mux(io.fn === aluFn.FN_SL,                           shout_l, 0.U)
 
+  // CZEQZ, CZNEZ
+  val cond_out = Option.when(coreParams.useConditionalZero)(
+    Mux(io.fn === aluFn.FN_CZEQZ && io.in2 =/= 0.U, io.in1, 0.U) |
+    Mux(io.fn === aluFn.FN_CZNEZ && io.in2 === 0.U, io.in1, 0.U)
```

Comment:
Too many MUXes and compare. 

```scala
val in2_not_zero = io.in2.orR
Mux(io.fn === aluFn.FN_CZEQZ && in2_not_zero || io.fn === aluFn.FN_CZNEZ && !in2_not_zero, io.in1, 0.U)
```

This should be enough

Reply:
@jerryz123 Are you happy with directly modifying ALU or should we add another ALU like ALUZicond? The main problem with the latter way is that then we will have a bunch of ALU  to maintain like `ALU`, `ALUZicond`, `ABLU`, `ABLUZicond`.

Reply:
This approach is fine.